### PR TITLE
Handle io.Reader to buffer errors

### PR DESCRIPTION
**Pull request by rjtsdl**

### CODE REVIEW

File: pkg/backup/abs/abs.go [hunk 1]
```diff
@@ -89,15 +90,23 @@ func (w *ABS) Put(key string, r io.Reader) error {
...
+		r = &buf
+	}
+
 	err := blob.CreateBlockBlob(&storage.PutBlobOptions{})
 	if err != nil {
 		return err
 	}
 
-	buf := new(bytes.Buffer)
-	buf.ReadFrom(r)
```
Comment (rjtsdl) [inline pkg/backup/abs/abs.go:98]: This is the reason we get partial bytes (or corrupted backup) from io.Reader. We forget to check the error of this function. 
The fix is above.

**Review by nicolerenee — APPROVED** [review]
👍 looks good

**Review by vdice — APPROVED** [review]

File: pkg/backup/abs/abs.go (hunk 1 above)
Comment (rite2nikhil) [reply pkg/backup/abs/abs.go]: Why do we need to copy to find the error ?

File: pkg/backup/abs/abs.go [hunk 2]
```diff
@@ -89,15 +90,23 @@ func (w *ABS) Put(key string, r io.Reader) error {
 	blobName := path.Join(v1, w.prefix, key)
 	blob := w.container.GetBlobReference(blobName)
 
+	var buf bytes.Buffer
+	if r != nil {
+		_, err := io.Copy(&buf, r)
```
Comment (rite2nikhil) [inline pkg/backup/abs/abs.go:95]: This will also return io.EOF error which is not an error, you should handle the error using ReadFrom instead

**Review by rite2nikhil — DISMISSED** [review]
Use error returned from ReadFrom instead

File: pkg/backup/abs/abs.go (hunk 2 above)
Comment (rjtsdl) [reply pkg/backup/abs/abs.go:95]: Check out here. https://github.com/Azure/azure-sdk-for-go/blob/master/storage/blockblob.go#L116-L133

This is how go sdk handle this stuff. I want to use their practice.

Comment (rite2nikhil) [reply pkg/backup/abs/abs.go:95]: Ok this is what readfrom does, I am ok if you want to copy what sdk does :)

**Pull request merged**